website/blog/2024-10-23-The-New-Architecture-is-Here.md

```diff
@@ -0,0 +1,397 @@
+
+This release is the result of a ground up rewrite of React Native we’ve been working on since 2018, and we’ve taken special care to make the New Architecture a gradual migration for most apps. In 2021, we created [the New Architecture Working Group](https://github.com/reactwg/react-native-new-architecture/) to collaborate with the community on ensuring a smooth upgrade experience for the entire React ecosystem.
+
+Most apps will be able to adopt React Native 0.76 with the same level of effort as any other release. The most popular React Native libraries already support the New Architecture. The New Architecture also includes an automatic interoperability layer to enable backward compatibility with libraries targeting the old architecture.
+
+<!--truncate-->
+
+Over the past several years of development, our team has publicly shared our vision for the New Architecture. If you missed any of these talks, check them out here:
+
+- [React Native EU 2019 \- The New React Native](https://www.youtube.com/watch?v=52El0EUI6D0)
+- [React Conf 2021 \- React 18 Keynote](https://www.youtube.com/watch?v=FZ0cG47msEk)
+- [App.js 2022 \- Bringing the New React Native Architecture to the OSS Community](https://www.youtube.com/watch?v=GJxL2aiIX3Q)
+- [React Conf 2024 \- Day 2 keynote](https://www.youtube.com/watch?v=Q5SMmKb7qVI)
+
+## What is the New Architecture
+
+The most important change in this release is something most users should not need to think about: how React Native works. We have completely rewritten the major systems of React Native including how components are rendered, how JavaScript communicates with native, and how work is scheduled across different threads. We call all of these architecture improvements the New Architecture.
+
+In the old architecture, React Native communicated with native with an asynchronous bridge. To render a component or call a native function, React Native needed to push native functions calls to a queue in the bridge, and process them asynchronously. This architecture had many benefits. Since all work was done on a background thread, the main thread was never blocked for rendering updates or handling Native Module function calls.
+
+The approach had major limitations. Some work needs to respond synchronously to user input, potentially interrupting any in progress work, in order to provide the best experience possible to users. Since the old architecture was only asynchronous, we needed to rewrite it to allow for both asynchronous and synchronous updates.
+
```

Comment:
Synchronous user input is not (yet) supported in the new architecture, so we don't have synchronous update. What we do have now is the ability for user input to interrupt other long-running tasks.

Reply:
It's nuanced, but this isn't talking specifically about dispatching events synchronously. What it does in the New Architecture today is that read renders discrete events synchronously (without yielding) interrupting lower priority updates like transitions. But I can tweak this to talk about rendering synchronously instead of events.

Reply:
Fixed in the other branch.